### PR TITLE
CDAP-12137 add null field splitter

### PR DESCRIPTION
**Pull request by albertshau**
Adding a SplitterTransform that has two output ports, one for
records with a null field value and one for records that have
a non-null value.

### CODE REVIEW

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/NullFieldSplitter.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    }
+    return Schema.recordOf(nullableSchema.getRecordName() + ".nonnull", fields);
+  }
+
+  /**
+   * Configuration for the plugin.
+   */
+  public static class Conf extends PluginConfig {
+    @Description("Which field should be checked for null values.")
+    private final String field;
```
Comment (gsps1) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/NullFieldSplitter.java]: should we allow this to be a macro ?

**Review by gsps1 — COMMENTED** [review]
just one question, LGTM otherwise.

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/NullFieldSplitter.java (hunk 1 above)
Comment (albertshau) [reply transform-plugins/src/main/java/co/cask/hydrator/plugin/NullFieldSplitter.java]: can make it macro enabled, will update.

**Comment by gsps1** [comment]
LGTM

**Pull request merged**